src/v2i-hub/CARMAStreetsPlugin/src/J2735MapToJsonConverter.cpp

```diff
@@ -0,0 +1,267 @@
+
+
+#include "J2735MapToJsonConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    void J2735MapToJsonConverter::convertJ2735MAPToMapJSON(const std::shared_ptr<MapData> mapMsgPtr, Json::Value &mapJson) const
+    {
+        // Construct metadata
+        Json::Value metadata;
+        auto timestamp_utc = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        metadata["timestamp"] = std::to_string(timestamp_utc);
+        mapJson["metadata"] = metadata;
+
+        // Construct Map Data
+        Json::Value mapDataJson;
+        mapDataJson["layer_id"] = mapMsgPtr->layerID;
+        mapDataJson["msg_issue_revision"] = std::to_string(mapMsgPtr->msgIssueRevision);
+        mapDataJson["layer_type"] = std::to_string(*mapMsgPtr->layerType);
```

Comment:
Why does the mapMsgPtr need to be dereferenced here and not in the line above?

Reply:
LayerType is a pointer